.github/workflows/validate-native-docker/native_docker_validator.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+import argparse
+import requests
+import os
+import sys
+
+
+DOCKER_FILES_SUFFIX = ["Dockerfile", "Pipfile", "Pipfile.lock"]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Deploy a pack from a contribution PR to a branch')
+    parser.add_argument('-c', '--changed_files', help='list of the changed files')
+    args = parser.parse_args()
+    changed_files = args.changed_files.split(" ")
+    return validate_native_docker(changed_files)
+
+
+def validate_native_docker(changed_files):
+    """
+    This function validate that no native docker supported dockers are being updated.
+    Args:
+        changed_files: the list of files that are being changed in the current pr.
+
+    Returns: the exit code according to wether the list of native docker supported dockers is empty or not.
+    """
+    updated_supported_dockers = get_updated_supported_dockers(changed_files)
+    if updated_supported_dockers:
+        print(f"the following dockers are updated and supported by the native docker: {','.join(updated_supported_dockers)}. Please make sure to update the native docker accordingly and add the 'docker update approved' label.")
+        return 1
+    return 0
+
+
+def get_updated_supported_dockers(changed_files):
+    """
+    This function list the dockers that are supported by the native docker and being updated at the current pr.
+    Args:
+        changed_files: the list of files that are being changed in the current pr.
+
+    Returns: the list of the native supported dockers supported dockers that are being updated.
+    """
+    updated_supported_dockers = []
+    updated_dockers_ls = get_list_of_updated_dockers(changed_files)
+    if updated_dockers_ls:
+        supported_dockers = get_supported_dockers()
+        for updated_docker in updated_dockers_ls:
+            if updated_docker in supported_dockers:
+                updated_supported_dockers.append(updated_docker)
+    return updated_supported_dockers
+
+
+def get_list_of_updated_dockers(changed_files):
+    """
+    This function list the dockers that are being updated at the current pr.
+    Args:
+        changed_files: the list of files that are being changed in the current pr.
+
+    Returns: the list of the dockers that are being updated at the current pr.
+    """
+    dockers = []
+    for file in changed_files:
+        for suffix in DOCKER_FILES_SUFFIX:
```

Comment:
dont think this is a great way to check. Some updates dont touch the dockerfile, sometimes we use poetry.

I think a better method would be to check if docker/{dockername}/xxx was changed. The folder name is really the id for the dockerfile, so we should check if anything in the folder was changed

Reply:
there're some files we wouldn't mind if they get change within the folder such as "build.conf" that's why I'm filtering by those we do, maybe I should add the poetry and pyproject files to this list?